core/src/main/kotlin/org/evomaster/core/problem/externalservice/httpws/HarvestActualHttpWsResponseHandler.kt

```diff
@@ -220,51 +218,54 @@ class HarvestActualHttpWsResponseHandler {
     /**
      * @return a copy of actual responses based on the given [httpRequest] and probability
      */
-    fun getACopyOfActualResponse(httpRequest: HttpExternalServiceRequest, probability: Double?=null) : ResponseParam?{
+    fun getACopyOfActualResponse(httpRequest: HttpExternalServiceRequest, probability: Double? = null): ResponseParam? {
         val harvest = probability == null || (randomness.nextBoolean(probability))
         if (!harvest) return null
-        synchronized(actualResponses){
-            val found= (actualResponses[httpRequest.getDescription()]?.param?.copy() as? ResponseParam)
-            if (found!=null) seededResponses.add(httpRequest.getDescription())
+        synchronized(actualResponses) {
+            val found = (actualResponses[httpRequest.getDescription()]?.param?.copy() as? ResponseParam)
+            if (found != null) seededResponses.add(httpRequest.getDescription())
             return found
         }
     }
 
     /**
      * add http request to queue for sending them to real external services
      */
-    fun addHttpRequests(requests: List<HttpExternalServiceRequest>){
-        if (requests.isEmpty())  return
+    fun addHttpRequests(requests: List<HttpExternalServiceRequest>) {
+        if (requests.isEmpty()) return
 
         if (!config.doHarvestActualResponse())
             return
 
         // only harvest responses with GET method
         //val filter = requests.filter { it.method.equals("GET", ignoreCase = true) }
-        synchronized(cachedRequests){
+        synchronized(cachedRequests) {
             requests.forEach { cachedRequests.putIfAbsent(it.getDescription(), it) }
         }
 
         addRequests(requests.map { it.getDescription() })
     }
 
-    private fun addRequests(requests : List<String>) {
+    private fun addRequests(requests: List<String>) {
         if (requests.isEmpty()) return
 
         val notInCollected = requests.filterNot { actualResponses.containsKey(it) }.distinct()
         if (notInCollected.isEmpty()) return
 
-        synchronized(lock){
+        synchronized(lock) {
             val newRequests = notInCollected.filterNot { queue.contains(it) }
             if (newRequests.isEmpty())
                 return
             updateExtractedObjectDto()
             lock.notify()
             queue.addAll(newRequests)
+
+            val task = Runnable { sendRequestToRealExternalService() }
+            workerPool.execute(task)
```

Comment:
it seems directly sending requests based on a sequence of the requests to be sent by SUT (ie, send the request after each addition).
then the queue seems less useful that might be problematic if we want to optimize them.

we might need to discuss it tomorrow.